code/modules/surgery/bodyparts/_bodyparts.dm

```diff
@@ -784,42 +798,94 @@
 	drop_organs()
 	return ..()
 
-/**
- * Calculates how much blood this limb is losing per life tick
- *
- * Arguments:
- * * ignore_modifiers - If TRUE, ignore the bleed reduction for laying down and grabbing your limb
- */
-/obj/item/bodypart/proc/get_part_bleed_rate(ignore_modifiers = FALSE)
+/// INTERNAL PROC, DO NOT USE
+/// Properly sets us up to manage an inserted embeded object
+/obj/item/bodypart/proc/_embed_object(obj/item/embed)
+	if(embed in embedded_objects) // go away
+		return
+	// We don't need to do anything with projectile embedding, because it will never reach this point
+	RegisterSignal(embed, COMSIG_ITEM_EMBEDDING_UPDATE, .proc/embedded_object_changed)
+	embedded_objects += embed
+	refresh_bleed_rate()
+
+/// INTERNAL PROC, DO NOT USE
+/// Cleans up any attachment we have to the embedded object, removes it from our list
+/obj/item/bodypart/proc/_unembed_object(obj/item/unembed)
+	UnregisterSignal(unembed, COMSIG_ITEM_EMBEDDING_UPDATE)
+	embedded_objects -= unembed
+	refresh_bleed_rate()
+
+/obj/item/bodypart/proc/embedded_object_changed(obj/item/embedded_source)
+	SIGNAL_HANDLER
```

Comment:
```suggestion
/obj/item/bodypart/proc/embedded_object_changed(obj/item/embedded_source)
	PROTECTED_PROC(TRUE)
	SIGNAL_HANDLER
```

Reply:
Why no should call parent here? what's the difference?

Reply:
PROTECTED_PROC means the proc cannot be called outside of bodypart code, mostly to prevent idiocy.

Reply:
do you think that's worth the extra line? I guess you would, since you made it.
as a general pattern I wouldn't like it, too much fluff

Reply:
see above one